server/honeybadger/plugins.py

```diff
@@ -26,18 +24,43 @@ def get_coords_from_google(aps):
         data['lng'] = jsondata['location']['lng']
     return data
 
-def get_coords_from_uniapple(ip):
-    logger.info('Geolocating via Uniapple API.')
-    url = 'http://uniapple.net/geoip/?ip={}'.format(ip)
-    content = urllib2.urlopen(url).read()
-    logger.info('Uniapple API response:\n{}'.format(content))
+def get_coords_from_ipstack(ip):
+    logger.info('Geolocating via Ipstack API.')
+    url = 'http://api.ipstack.com/{0}?access_key={1}'.format(ip, app.config['IPSTACK_API_KEY'])
+    content = requests.get(url).content
+    logger.info('Ipstack API response:\n{}'.format(content))
     jsondata = None
     try:
         jsondata = json.loads(content)
     except ValueError as e:
         logger.error('{}.'.format(e))
+
+    # Avoid the KeyError. For some reason, a successful API call to Ipstack doesn't include
+    #   the 'success' key in the json result, but a failed call does, and the value is False
+    if 'success' in jsondata and not jsondata['success']:
+        logger.info('Ipstack API call failed: {}'.format(jsondata['error']['type']))
+        # Return with no data so the caller knows to default to the fallback API
+        return None
```

Comment:
Just happened to think on this review, is this the right response? All of the others return a structured dictionary with the values set to `None`. How does the calling code handle a NoneType response like this?

Reply:
The calling code checks for this specific method returning None, and then hands it off to the additional fallback. Previously, as there was only one fallback, this logic was not necessary. This can be restructured if you think it's warranted.

Reply:
Unfortunately, I think I need to look at this a bit deeper when I have more time.

Reply:
Fair enough. I'll also take a look and see about making it consistent with the other functions.

Reply:
I've changed this to match the style of the other functions.